src/main/java/ch/njol/skript/expressions/ExprAdventureRestrictions.java

```diff
@@ -0,0 +1,177 @@
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.Material;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+@Name("Adventure Restrictions Of Item")
+@Description("Get or modify the adventure restrictions of an item.")
+@Examples({
+	"add dirt to destroyable restrictions of player's tool",
+	"add (stone, diamond ore) to breakable blocks of {_item}",
+	"clear break restrictions of {_item}",
+	"remove sand from destroyable blocks of {_item} in adventure mode"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
+public class ExprAdventureRestrictions extends PropertyExpression<ItemType, ItemType> {
+
+	static {
+		if (Skript.methodExists(ItemMeta.class, "getDestroyableKeys")) {
+			Skript.registerExpression(ExprAdventureRestrictions.class, ItemType.class, ExpressionType.PROPERTY,
+				"[the] (:break|:destroy|:place|:build)able blocks of %itemtypes% [in adventure [mode]]",
+				"[the] (:break|:destroy|:place|:build)[able] restrictions of %itemtypes%",
+				"%itemtypes%'[s] (:break|:destroy|:place|:build)able blocks [in adventure [mode]]",
+				"%itemtypes%'[s] (:break|:destroy|:place|:build)[able] restrictions");
+		}
+	}
+
+	private boolean destroy;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<ItemType>) exprs[0]);
+		this.destroy = parseResult.hasTag("break") || parseResult.hasTag("destroy");
+		return true;
+	}
+
+	@Override
+	protected ItemType @NotNull [] get(Event event, ItemType [] source) {
+		Set<ItemType> existingKeys = new HashSet<>();
+		for (ItemType item : source) {
+			ItemMeta meta = item.getItemMeta();
+			if (destroy ? meta.hasDestroyableKeys() : meta.hasPlaceableKeys()) {
+				for (Namespaced key : destroy ? meta.getDestroyableKeys() : meta.getPlaceableKeys()) {
+					Material material = BukkitUnsafe.getMaterialFromMinecraftId(key.toString());
+					if (material != null) {
+						existingKeys.add(new ItemType(material));
+					}
+				}
+			}
+		}
+		return existingKeys.toArray(new ItemType[0]);
+	}
+
+	@SuppressWarnings("NullableProblems")
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) {
+			return CollectionUtils.array(ItemType[].class);
+		} else if (mode == ChangeMode.RESET || mode == ChangeMode.DELETE) {
+			return CollectionUtils.array();
+		}
+		return null;
```

Comment:
might be nicer as a switch, completely up to you

Reply:
I think I'd rather leave this as a couple if's, but happy to change it if need be